pkg/messaging/pair/client.go

```diff
@@ -10,20 +10,20 @@ import (
 	"go.nanomsg.org/mangos/v3/protocol/pair"
 )
 
-func StartMessagingPairClient(ctx context.Context, nanomsgURL string, requestPair chan RequestPair, responsePair chan ResponsePair) error {
-	socket, err := pair.NewSocket()
+func StartPairMessagingClient(ctx context.Context, nanomsgURL string, requestPair chan RequestPair, responsePair chan ResponsePair) error {
```

Comment:
Add channel flow annotation (`chan <-` or `<- chan`) to this signature. Please, find all places where you can see the same problem and fix them.